src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.c

```diff
@@ -61,25 +76,999 @@ static ColliderCylinderInit D_809D5C40 = {
+                    Math_ApproachF(&this->cutsceneCameraEye.x, cutsceneCameraEyeTarget.x, 0.2f, 500.0f);
+                    Math_ApproachF(&this->cutsceneCameraEye.y, cutsceneCameraEyeTarget.y, 0.2f, 500.0f);
+                    Math_ApproachF(&this->cutsceneCameraEye.z, cutsceneCameraEyeTarget.z, 0.2f, 500.0f);
+                    Gameplay_CameraSetAtEye(globalCtx2, this->cameraId, &this->cutsceneCameraEye,
+                                            &this->cutsceneCameraAt);
+                }
+
+                // Cutscene has finished.
+                if (this->cutsceneTimer == 1) {
+                    func_800C08AC(globalCtx2, this->cameraId, 0);
+                    this->cutsceneMode = this->cameraId = 0;
+                    func_80064534(globalCtx2, &globalCtx2->csCtx);
+                }
+            }
+        }
+    }
+
+    // Explode the Arwing
+    if (this->shouldExplode) {
+        this->shouldExplode = 0;
+        Audio_PlaySoundAtPosition(globalCtx2, &this->actor.world.pos, 40, NA_SE_IT_BOMB_EXPLOSION);
+
+        // Spawn flash effect.
+        crashEffectLocation.x = this->actor.world.pos.x;
+        crashEffectLocation.y = (this->actor.world.pos.y + 40.0f) - 30.0f;
+        crashEffectLocation.z = this->actor.world.pos.z;
+        EnClearTag_CreateFlashEffect(globalCtx2, &crashEffectLocation, 6.0f, this->actor.floorHeight,
+                                     &this->floorTangent);
+
+        // Spawn smoke effect.
+        crashEffectLocation.y = (this->actor.world.pos.y + 30.0f) - 50.0f;
+        EnClearTag_CreateSmokeEffect(globalCtx2, &crashEffectLocation, 3.0f);
+        crashEffectLocation.y = this->actor.world.pos.y;
+
+        // Spawn debris effects.
+        for (i = 0; i < 15; i++) {
+            crashEffectVelocity.x = sinf(i * 1.65f) * i * 0.3f;
+            crashEffectVelocity.z = cosf(i * 1.65f) * i * 0.3f;
+            crashEffectVelocity.y = Rand_ZeroFloat(6.0f) + 5.0f;
+            crashEffectVelocity.x += Rand_CenteredFloat(0.5f);
+            crashEffectVelocity.z += Rand_CenteredFloat(0.5f);
+
+            debrisEffectAcceleration.x = 0.0f;
+            debrisEffectAcceleration.z = 0.0f;
+            debrisEffectAcceleration.y = -1.0f;
+
+            EnClearTag_CreateDebrisEffect(globalCtx2, &crashEffectLocation, &crashEffectVelocity,
+                                          &debrisEffectAcceleration, Rand_ZeroFloat(0.15f) + 0.075f,
+                                          this->actor.floorHeight);
+        }
+    }
+
+    if (this->drawMode != CLEAR_TAG_DRAWMODE_MODEL) {
+        // Check if the Arwing should be removed.
+        if (this->drawMode == CLEAR_TAG_DRAWMODE_EFFECT) {
+            if (this->deathTimer == 0) {
+                isArwingAlive = 0;
+            } else {
+                this->deathTimer--;
+                isArwingAlive = this->deathTimer;
+            }
+
+            if (!isArwingAlive) {
```

Comment:
This looks like `if(DECR(this->deathTimer) == 0) {`

Reply:
I've tried DECR in the file and it's not OK.